velox/functions/prestosql/fuzzer/ApproxPercentileResultVerifier.h

```diff
@@ -88,7 +89,7 @@ class ApproxPercentileResultVerifier : public ResultVerifier {
         valueField,
         weightField,
         sortingKeysAndOrders,
-        function.frame,
+        frameClause,
```

Comment:
What's the difference between function.frame and frameClause? Also, why does ApproxPercentileResultVerifier use frameClause but ApproxDistinctResultVerifier uses frame? Is there a difference between the two verifiers that causes this distinction?

Reply:
In `ApproxPercentileResultVerifier`, the frame clause is constructed using `function.frame` with the helper function:
```
std::string getFrameClause(const core::WindowNode::Frame& frame);
```
This function was removed as it doesn't account for frames with columns as bounds, and resulted in the error shared before. Also since `frameClause` is now passed as an argument to `verifyWindow`, it could be passed directly to `initializeWindow` in the verifier. The usage in `ApproxPercentileResultVerifier`:
```
.window({fmt::format(
            "multimap_agg(x, w) over ({} {} {}) as bucket",
            getPartitionByClause(groupingKeys_),
            getOrderByClause(sortingKeysAndOrders),
            frameClause)})
```

In `ApproxDistinctResultVerifier` on the other hand, the `over clause` in the window is used, this is passed in the argument `frame`:
```
auto plan = PlanBuilder()
                    .values(input)
                    .window({makeCountDistinctWindowCall(function, frame)})
                    .planNode();
```
Please let me know if the variable names should be changed or other refactor is needed.

Reply:
Hi @pramodsatya, Got it. Thank you for the detailed explanation. I found the reason why getFrameClause() doesn't work for column bounds: getFrameClause() uses WindowNode::Frame::startValue->toString() while startValue is a FieldAccessTypedExpr when it's a column bound. FieldAccessTypedExpr::toString() returns `ROW["off1"]` instead of the `off1` we want.

```
.window({fmt::format(
            "multimap_agg(x, w) over ({} {} {}) as bucket",
            getPartitionByClause(groupingKeys_),
            getOrderByClause(sortingKeysAndOrders),
            frameClause)})
```
This fmt::format() actually generates the same string as the `frame` argument of initializeWindow(). An easy solution is to pass this `frame` to computePercentilesForWindow(), then replace the statement above by the following.
```
.window({fmt::format(
            "multimap_agg(x, w) over ({}) as bucket",
            frame)})
```

Then you don't need to pass an additional argument `frameClause` to initializeWindow() and can remove the getPartitionByClause() and getOrderByClause(). In general, we'd like to avoid passing multiple arguments with redundant information because that could cause confusion.

Reply:
Thanks for the feedback @kagamiori. I wasn't sure if the fmt::format output was same as `frame`, updated to use the existing argument as suggested.